src/workerinfo.js

```diff
@@ -59,12 +61,43 @@ class WorkerInfo {
     });
   }
 
+  async workerTypeSeen(provisionerId, workerType) {
```

Comment:
Maybe `WorkerInfo` just has `WorkerInfo.seen(provisionerId, workerType)` which then in turn calls the two different seen functions concurrently...